logstash-core/lib/logstash/webserver.rb

```diff
@@ -80,6 +83,18 @@ def stop(options={})
 
     private
 
+    def pick_default_port(http_host, range=(9600..9700))
+      range.step(1) do |current_port|
+        begin
+          TCPServer.new(http_host, current_port).close
```

Comment:
hmm, this will not work.. isn't there a race condition across 2 instance? What if LS instance A checked the port for free-ness and returned ok, but before it actually acquires the port the other LS instance does the same? I think the check and acquire port should happen at the same time..